### PR TITLE
Refactor argument parser construction

### PR DESCRIPTION
**Pull request by clbarnes**
Each subparser is added in its own function

### CODE REVIEW

**Comment by clbarnes** [comment]
Just checked and it's mergeable with the pipxrc PR.

File: pipx/main.py [hunk 1]
```diff
@@ -30,6 +31,7 @@
 
 
 __version__ = "0.14.0.0"
+__version_info__ = tuple(int(n) for n in __version__.split("."))
```
Comment (cs01) [inline pipx/main.py]: I sometimes make beta releases, such as ` "0.14.1.0b0"`. Can the `int` cast be omitted?

Comment (clbarnes) [reply pipx/main.py]: The intention of `__version_info__` is to sort properly, which a double-digit string wouldn't do - if 4 segments and an optional beta segment is the only case to account for I don't mind making this logic more complete.

**Comment by cs01** [comment]
I updated pipx's unit tests and some of pipx's code, but now there is a merge conflict in this PR. But it should be more straightforward to write tests now and measure coverage.

File: pipx/main.py [hunk 2]
```diff
@@ -512,11 +562,11 @@ def setup(args):
     else:
         logging.basicConfig(level=logging.WARNING, format="%(message)s")
 
-    mkdir(constants.PIPX_LOCAL_VENVS)
-    mkdir(constants.LOCAL_BIN_DIR)
-    mkdir(constants.PIPX_VENV_CACHEDIR)
+    mkdir(PIPX_LOCAL_VENVS)
```
Comment (cs01) [inline pipx/main.py]: The tests patch the value of these constants, so they need the `constants.` prefix

File: pipx/main.py [hunk 3]
```diff
@@ -6,20 +6,27 @@
...
 from .util import PipxError, mkdir
 from .Venv import VenvContainer
 
+
 __version__ = "0.14.0.0"
+_, __version_info__, subver, *_ = parse_version(__version__)._key  # type: ignore
+if isinstance(subver, tuple):
+    while len(__version_info__) < 4:
+        __version_info__ += (0,)
+    __version_info__ += subver
```
Comment (cs01) [inline pipx/main.py]: would you mind adding a unit test for this?

Comment (clbarnes) [reply pipx/main.py]: Sure. Are we set on only allowing a version with 1 to 4 components, plus an optional a/b/c component?

Comment (cs01) [reply pipx/main.py]: Yeah it will always be something like 0.1.2.3.4, 0.1.2.3.4b0, 0.1.2.3.4rc0, etc.

File: pipx/main.py [hunk 4]
```diff
@@ -257,21 +271,7 @@ def _autocomplete_list_of_installed_packages(
...
-
-    parser = argparse.ArgumentParser(
-        formatter_class=LineWrapRawTextHelpFormatter, description=PIPX_DESCRIPTION
-    )
-
-    subparsers = parser.add_subparsers(
-        dest="command", description="Get help for commands with pipx COMMAND --help"
-    )
-
+def _add_install(subparsers):
```
Comment (cs01) [inline pipx/main.py]: can all these functions have types added to them?

**Comment by cs01** [comment]
Looks good, thanks! I'll merge now and we can add types later if we want.

**Pull request merged**